src/workflows/airqo_etl_utils/datautils.py

```
@@ -204,9 +204,26 @@ def extract_devices_data(
+        d_category = None
+
         devices_data = pd.DataFrame()
+        if device_category == DeviceCategory.MOBILE:
+            d_category = DeviceCategory.MOBILE
+            device_category = DeviceCategory.LOWCOST
 
         devices, keys = DataUtils.get_devices(device_category, device_network)
+        # Temporary fix for mobile devices
+        # TODO: Fix after requirements review
+        if d_category:
+            if "assigned_grid" in devices.columns:
```

Ruff (0.12.2)

Use a single `if` statement instead of nested `if` statements

Combine `if` statements using `and`

(SIM102)